src/NhanAZ/DataCleaner/Main.php

```diff
@@ -10,96 +10,117 @@
 
 class Main extends PluginBase {
 
-	// Sorry Poggit Reviewers, The code of this plugin is not clean!
-
-	private array $deletedData = [];
-
-	private function getDataPath() {
-		return $this->getServer()->getDataPath() . "plugin_data" . DIRECTORY_SEPARATOR;
+	private function getExceptionData(): array {
+		return array_merge($this->getConfig()->get("exceptionData", []), [".", ".."]);
 	}
 
-	private function getExceptionData(): array {
-		$exceptionData = $this->getConfig()->get("exceptionData");
-		return $exceptionData[] = [".", ".."];
+	private function deleteMessage(array $deleted): void {
+		$this->getLogger()->info("§fDeleted data (" . count($deleted) . "): §a" . implode("§f,§a ", $deleted));
 	}
 
-	private function deleteMessage() {
-		$deletedData = implode("§f,§a ", $this->deletedData);
-		$dataCount = count($this->deletedData);
-		$deleteMessage = "§fDeleted data ($dataCount): §a$deletedData";
-		$this->getLogger()->info($deleteMessage);
+	/**
+	 * @param bool $justEmpty (only for folders) true delete the folder only if it is empty or false delete the folder and all its contents
+	 *
+	 * @return bool true on success or false on failure.
+	 */
+	public function delete(\DirectoryIterator $fileInfo, bool $justEmpty = false): bool {
+		if ($fileInfo->isDir()) {
+			return $this->deleteFolder($fileInfo, $justEmpty);
```

Comment:
why not just use [`Filesystem::recursiveUnlink`](https://github.com/pmmp/PocketMine-MP/blob/fdb3a5b121e8064f099ad1d48aa1c0e84103a2df/src/utils/Filesystem.php#L81)?